extensions/default/src/DicomJSONDataSource/index.js

```diff
@@ -258,19 +262,19 @@ function createDicomJSONApi(dicomJsonConfig) {
 
       const { StudyInstanceUID, SeriesInstanceUID } = displaySet;
       const study = findStudies('StudyInstanceUID', StudyInstanceUID)[0];
-      const series = study.series.find(s => s.SeriesInstanceUID === SeriesInstanceUID);
-      let instances = displaySet.images;
-      if (series.instances.length > displaySet.images.length) {
-        instances = series.instances;
-      }
+      const series = study.series.find(s => s.SeriesInstanceUID === SeriesInstanceUID) || [];
 
-      instances.forEach(instance => {
+      displaySet.images.forEach(instance => {
         const NumberOfFrames = instance.NumberOfFrames;
-
         if (NumberOfFrames > 1) {
           for (let i = 0; i < NumberOfFrames; i++) {
+            /** In case there are multiple sop class we filter sop instances */
+            const images = series.instances
+              .filter(i => i && i.metadata && i.metadata.SOPInstanceUID === instance.SOPInstanceUID)
+              .map(i => ({ ...(i.metadata || {}), url: i.url }));
```

Comment:
This will copy the metadata multiple times, which can get very expensive for large multiframes.  I have examples with 340,000 frames in it, and those should still work successfully.

Reply:
Updated!